contracts/exchange/test/utils/match_order_tester.ts

```diff
@@ -634,4 +910,61 @@ function encodeTokenBalances(obj: any): any {
     const keys = _.keys(obj).sort();
     return _.zip(keys, keys.map(k => encodeTokenBalances(obj[k])));
 }
+
+/**
+ * Gets the last match in a BatchMatchResults object.
+ * @param batchMatchResults The BatchMatchResults object.
+ * @return The last match of the results.
+ */
+function getLastMatch(batchMatchResults: BatchMatchResults): MatchResults {
+    return batchMatchResults.matches[batchMatchResults.matches.length - 1];
+}
+
+/**
+ * Get the token balances
+ * @param batchMatchResults The results of a batch order match
+ * @return The token balances results from after the batch
+ */
+function getUpdatedBalances(batchMatchResults: BatchMatchResults, initialTokenBalances: TokenBalances): TokenBalances {
+    return batchMatchResults.matches
+        .map(match => match.balances)
+        .reduce((totalBalances, balances) => aggregateBalances(totalBalances, balances, initialTokenBalances));
+}
+
+/**
+ * Takes a `totalBalances`, a `balances`, and an `initialBalances`, subtracts the `initialBalances
+ * from the `balances`, and then adds the result to `totalBalances`.
+ * @param totalBalances A set of balances to be updated with new results.
+ * @param balances A new set of results that deviate from the `initialBalances` by one matched
+ *                 order. Subtracting away the `initialBalances` leaves behind a diff of the
+ *                 matched orders effect on the `initialBalances`.
+ * @param initialBalances The token balances from before the call to `batchMatchOrders()`.
+ * @return The updated total balances using the derived balance difference.
+ */
+function aggregateBalances(
+    totalBalances: TokenBalances,
+    balances: TokenBalances,
+    initialBalances: TokenBalances,
+): TokenBalances {
+    // ERC20
+    for (const owner of _.keys(totalBalances.erc20)) {
+        for (const contract of _.keys(totalBalances.erc20[owner])) {
+            const difference = balances.erc20[owner][contract].minus(initialBalances.erc20[owner][contract]);
+            totalBalances.erc20[owner][contract] = totalBalances.erc20[owner][contract].plus(difference);
+        }
+    }
+    // ERC721
+    for (const owner of _.keys(totalBalances.erc721)) {
+        for (const contract of _.keys(totalBalances.erc721[owner])) {
+            totalBalances.erc721[owner][contract] = _.zipWith(
+                totalBalances.erc721[owner][contract],
+                balances.erc721[owner][contract],
+                initialBalances.erc721[owner][contract],
+                (a: BigNumber, b: BigNumber, c: BigNumber) => a.plus(b.minus(c)),
+            );
+        }
+    }
+    // TODO(jalextowle): Implement the same as the above for ERC1155
```

Comment:
Do we have any `matchOrders` coverage for 1155s? Let's add an asana task for this.

Reply:
I'll check and get back to you.